pkg/transport/kafka-client/kafka_producer_client.go

```diff
@@ -0,0 +1,118 @@
+package kafkaclient
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/go-logr/logr"
+	kclient "github.com/open-cluster-management/hub-of-hubs-kafka-transport/kafka-client/kafka-producer"
+	"github.com/open-cluster-management/hub-of-hubs-spec-transport-bridge/pkg/transport"
+)
+
+// NewHOHProducer returns a new instance of HOHProducer object.
+func NewHOHProducer(log logr.Logger) (*HOHProducer, error) {
+	kp := &HOHProducer{
+		deliveryChan:  make(chan kafka.Event),
+		stopChan:      make(chan struct{}, 1),
+		kafkaProducer: nil,
+		log:           log,
+	}
+
+	kafkaProducer, err := kclient.NewKafkaProducer(kp.deliveryChan)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kafka producer: %w", err)
+	}
+
+	kp.kafkaProducer = kafkaProducer
+
+	return kp, nil
+}
+
+// HOHProducer abstracts hub-of-hubs-kafka-transport kafka-producer's generic usage.
+type HOHProducer struct {
+	log           logr.Logger
+	kafkaProducer *kclient.KafkaProducer
+	deliveryChan  chan kafka.Event
+	stopChan      chan struct{}
+	startOnce     sync.Once
+	stopOnce      sync.Once
+}
+
+// deliveryHandler handles results of sent messages.
+// For now failed messages are only logged.
+func (p *HOHProducer) deliveryHandler(e *kafka.Event) {
+	switch ev := (*e).(type) {
+	case *kafka.Message:
+		if ev.TopicPartition.Error != nil {
+			load := &transport.Message{}
+
+			err := json.Unmarshal(ev.Value, load)
+			if err != nil {
+				p.log.Error(err, "Failed to deliver message",
+					"Topic Name", ev.TopicPartition)
+				return
+			}
+
+			p.log.Error(ev.TopicPartition.Error, "Failed to deliver message",
+				"Message ID", load.ID, "Topic Name", ev.TopicPartition)
+		}
+	default:
+		p.log.Info("Received unsupported kafka-event type", "Message Type", ev)
+	}
+}
+
+// Start starts the kafka-client.
+func (p *HOHProducer) Start() {
+	p.startOnce.Do(func() {
+		// Delivery report handler for produced messages
+		go func() {
+			for {
+				select {
+				case <-p.stopChan:
+					return
+				case e := <-p.deliveryChan:
+					p.deliveryHandler(&e)
+				}
+			}
+		}()
+	})
+}
+
+// Stop stops the kafka-client.
+func (p *HOHProducer) Stop() {
+	p.stopOnce.Do(func() {
+		p.kafkaProducer.Close()
+		p.stopChan <- struct{}{}
+		close(p.stopChan)
+		close(p.deliveryChan)
+	})
+}
+
+// SendAsync sends a message to the sync service asynchronously.
+func (p *HOHProducer) SendAsync(id string, msgType string, version string, payload []byte) {
```

Comment:
if we want to keep the same structure as sync service the sendAsync function can only push to a msgChan channel and a different go routine "distributeMessages" should read from the channel and distribute it. 

what will happen if kafkaProducer is not responding? will your code hang?
if that's the case than it's a must to separate to different functions. 
we can't say we sendAsync but then hang.

if it will not hang, it's not a necessary step but still would be nicer to have same structured code.